index.js

```diff
@@ -33,40 +31,23 @@ module.exports = function (archive, target, opts, cb) {
   var watcher
 
   if (watch && archive.live) {
-    watcher = chokidar.watch([target], {
-      persistent: true,
-      ignored: opts.ignore
-    })
-    watcher.once('ready', function () {
-      watcher.on('add', function (file, stat) {
-        status.emit('file watch event', {path: file, mode: 'created'})
-        consume(file, stat)
-      })
-      watcher.on('change', function (file, stat) {
-        status.emit('file watch event', {path: file, mode: 'updated'})
-        consume(file, stat)
-      })
-      watcher.on('unlink', noop) // TODO
-    })
+    watcher = recursiveWatch(target, consume)
   }
 
   var status = new EventEmitter()
-  status.close = function () { watcher && watcher.close() }
+  status.close = function () { watcher && watcher() }
   status.fileCount = 0
   status.totalSize = 0
   status.bytesImported = 0
 
-  function consume (file, stat, cb) {
+  function consume (file, cb) {
     cb = cb || emitError
     if (opts.ignore && match(opts.ignore, file)) return cb()
-    if (stat) {
+    status.emit('file watch event', {path: file, mode: 'updated'})
```

Comment:
the `file watch event` would be called only when the file watcher triggered a change, not when it was read completely yet iirc. @joehand do you see a way how to best integrate this with the new module?